### PR TITLE
Make it possible to display absolute time stamps (since midnight)

### PR DESCRIPTION
**Pull request by c3d**
Three tweaks were added:

* `recorder_order` enables / disable the display of the order
* `recorder_abstime` enables / disables the display of absolute time
* `recorder_reltime` enables / dsiables the display of relative time

Signed-off-by: Christophe de Dinechin <dinechin@redhat.com>
Suggested-by: Frediano Ziglio <fzliglio@redhat.com>

### CODE REVIEW

File: recorder.c [hunk 1]
```diff
@@ -1965,7 +2003,15 @@ uintptr_t recorder_tick(void)
     uintptr_t tick = t.tv_sec * 1000ULL + t.tv_usec / 1000;
 #endif
     if (!initialTick)
+    {
         initialTick = tick;
+#if RECORDER_64BIT
+        recorder_time_at_start = tick % (86400 * 1000000ULL);
```
Comment (freddy77) [inline recorder.c]: why not use RECORDER_HZ instead?

File: recorder.c [hunk 2]
```diff
@@ -799,6 +813,16 @@ static void recorder_format_entry(recorder_show_fn show,
     char *dst = buffer;
     char *dst_end = buffer + sizeof buffer;
 
+#define rsnprintf(...)                                                  \
+    snprintf(dst, dst_end >= dst ? dst_end - dst : 0,  __VA_ARGS__)
+
+#if RECORDER_64BIT
+#define REC_TS_FMT  "%.6f"
```
Comment (freddy77) [inline recorder.c]: you could have instead

#if RECORDER_HZ == 1000000
#define REC_TS_FMT  "%.6f"
#elif RECORDER_HZ == 1000
#define REC_TS_FMT  "%.3f"
#else 
#error RECORDER_HZ not supported
#endif

File: recorder.c [hunk 3]
```diff
@@ -120,6 +120,25 @@ enum
 RECORDER_TWEAK_DEFINE(recorder_signals_mask,
                       RECORDER_SIGNALS_MASK,
                       "Recorder default mask for signals to catch");
+RECORDER_TWEAK_DEFINE(recorder_dump_sleep, 100,
```
Comment (freddy77) [inline recorder.c:132]: maybe these move should be split in a separate commit?

**Comment by freddy77** [comment]
typo "dsiables" in commit message

File: recorder.c (hunk 1 above)
Comment (c3d) [reply recorder.c]: Done

File: recorder.c (hunk 2 above)
Comment (c3d) [reply recorder.c]: Good idea. I went even a bit further after remembering that the code was designed to allow someone to pass a `RECORDER_HZ` at build time.

File: recorder.c (hunk 3 above)
Comment (c3d) [reply recorder.c:132]: Yes. I also split the rsnprintf part.
